zip-0244.rst

```diff
@@ -1,8 +1,382 @@
+.......................................
+A BLAKE2b-256 hash of the field encoding of all nullifier field
+values of Sapling shielded spends belonging to the transaction.
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSSpendCHash"
+
+3a.ii: ``sapling_spends_noncompact_digest``
+...........................................
+A BLAKE2b-256 hash of the non-nullifier information for all Sapling shielded spends
+belonging to the transaction. For each spend, the following elements are included
+in the hash:: 
+
+   * 3a.ii.1 ``cv``      (field encoding bytes)
+   * 3a.ii.2 ``anchor``  (field encoding bytes)
+   * 3a.ii.3 ``rk``      (field encoding bytes)
+   * 3a.ii.4 ``zkproof`` (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSSpendNHash"
+
+3b: ``sapling_outputs_digest``
+'''''''''''''''''''''''''''''''
+This digest is a BLAKE2b-256 hash of the following values ::
+
+   * 3a.i.   ``sapling_outputs_compact_digest`` (32-byte hash)
+   * 3b.ii.  ``sapling_outputs_memos_digest`` (32-byte hash)
+   * 3b.iii. ``sapling_outputs_noncompact_digest`` (32-byte hash)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutputHash"
+
+3b.i: ``sapling_outputs_compact_digest``
+.........................................
+A BLAKE2b-256 hash of the subset of Sapling output information included in the 
+ZIP-307 ``CompactBlock`` format for all Sapling shielded outputs
+belonging to the transaction. For each output, the following elements are included
+in the hash:: 
+
+   * 3b.i.1 ``cmu``                  (field encoding bytes)
+   * 3b.i.2 ``ephemeral_key``        (field encoding bytes)
+   * 3b.i.3 ``enc_ciphertext[..52]`` (First 52 bytes of field encoding)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutC__Hash"
+
+3a.ii: ``sapling_outputs_memos_digest``
+........................................
+A BLAKE2b-256 hash of the subset of Sapling shielded memo field data for all Sapling 
+shielded outputs belonging to the transaction. For each output, the following elements 
+are included in the hash:: 
+
+   * 3b.ii.1 ``enc_ciphertext[52..564] (contents of the encrypted memo field)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutM__Hash"
+
+3a.iii: ``sapling_outputs_noncompact_digest``
+..............................................
+A BLAKE2b-256 hash of the remaining subset of Sapling output information **not** included
+in the ZIP-307 ``CompactBlock`` format, for all Sapling shielded outputs belonging to the
+transaction. For each output, the following elements are included in the hash:: 
+
+   * 3b.iii.1 ``cv``                    (field encoding bytes)
+   * 3b.iii.2 ``enc_ciphertext[564..]`` (post-memo suffix of field encoding)
+   * 4b.iii.3 ``out_ciphertext``        (field encoding bytes)
+   * 4b.iii.4 ``zkproof``               (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutN__Hash"
```

Comment:
```suggestion
  "ZTxIdSOutN__Hash" (2 underscore characters)
```

Reply:
Using multiple underscores seems error-prone, can we use a number instead?